File: Easy/Roman To Integer/romanToInt.py
class Solution(object):
    def romanToInt(self, s):
        """
        :type s: str
        :rtype: int
        """
        roman_to_int = {
            'I': 1,
            'V': 5,
            'X': 10,
            'L': 50,
            'C': 100,
            'D': 500,
            'M': 1000
        }
        
        total = 0
        previous_val = 0

        for char in reversed(s):
            current_val = roman_to_int[char]
            if current_val < previous_val:
                total -= current_val
            else: 
                total += current_val
            previous_val = current_val
        return total

#QED
#Problem 13 (Roman To Integer) - Jason Balayev (python)   

File: Easy/Roman To Integer/romanToInt.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public int romanToInt(String s) {
        Map<Character, Integer> romanToInt = new HashMap<>();
        romanToInt.put('I',1);
        romanToInt.put('V',5);
        romanToInt.put('X',10);
        romanToInt.put('L',50);
        romanToInt.put('C',100);
        romanToInt.put('D',500);
        romanToInt.put('M',1000);

        int total = 0;
        int previousVal = 0;

        for (int i = s.length()-1; i>=0; i--) {
            int currentVal = romanToInt.get(s.charAt(i));
            if (currentVal<previousVal) {
                total -= currentVal;
            } else {
                total += currentVal;
            }
            previousVal = currentVal;
        }
        return total;
    }
}

//QED
//Problem 13 (Roman To Integer) - Jason Balayev (java)